include/openmc/cell.h

```diff
@@ -130,21 +133,54 @@ class Cell
   //! \param on_surface The signed index of a surface that the coordinate is
   //!   known to be on.  This index takes precedence over surface sense
   //!   calculations.
+  virtual bool
+  contains(Position p, Direction u, int32_t on_surface) const = 0;
+
+  //! Find the oncoming boundary of this cell.
+  virtual std::pair<double, int32_t>
+  distance(Position p, Direction u, int32_t on_surface) const = 0;
+
+  //! Write all information needed to reconstruct the cell to an HDF5 group.
+  //! @param group_id An HDF5 group id.
+  virtual void to_hdf5(hid_t group_id) const = 0;
+
+};
+ 
+class CSGCell : public Cell
+{
+public:
+
+  CSGCell();
+  
+  explicit CSGCell(pugi::xml_node cell_node);
+
   bool
   contains(Position r, Direction u, int32_t on_surface) const;
 
-  //! Find the oncoming boundary of this cell.
   std::pair<double, int32_t>
   distance(Position r, Direction u, int32_t on_surface) const;
 
-  //! \brief Write cell information to an HDF5 group.
-  //! \param group_id An HDF5 group id.
   void to_hdf5(hid_t group_id) const;
 
 protected:
   bool contains_simple(Position r, Direction u, int32_t on_surface) const;
   bool contains_complex(Position r, Direction u, int32_t on_surface) const;
 };
 
+#ifdef DAGMC
+class DAGCell : public Cell
+{
+ public:
+  moab::DagMC *dagmc_ptr_;
```

Comment:
Bummed I missed this one because I prefer that layout too... Fixing.

Our model in the past has been non-owned pointers which are cleaned up elsewhere at the end of the run. I'm open to other options though!